x-pack/elastic-agent/magefile.go

```diff
@@ -566,10 +622,34 @@ func runAgent(env map[string]string) error {
 }
 
 func packageAgent(requiredPackages []string, packagingFn func()) {
-	version, found := os.LookupEnv("BEAT_VERSION")
-	if !found {
-		version = release.Version()
+	fmt.Println("--- Package Elastic-Agent")
+	var packageVersion string
+	// if we have defined a manifest URL to package Agent from, we should be using the same packageVersion of that manifest
+	if devtools.PackagingFromManifest {
+		fmt.Println(">>>> Using manifest to package Agent")
+		if manifestResponse, err := manifest.DownloadManifest(devtools.ManifestURL); err != nil {
+			log.Panicf("failed to download remote manifest file %s", err)
+		} else {
+			if parsedVersion, err := version.ParseVersion(manifestResponse.Version); err != nil {
+				log.Panicf("the manifest version from manifest is not semver, got %s", manifestResponse.Version)
+			} else {
+				// When getting the packageVersion from snapshot we should also update the env of SNAPSHOT=true which is
+				// something that we use as an implicit parameter to various functions
+				if parsedVersion.IsSnapshot() {
+					os.Setenv(snapshotEnv, "true")
+					mage.Snapshot = true
+				}
+				os.Setenv("BEAT_VERSION", parsedVersion.CoreVersion())
```

Comment:
Why does the 7.17 branch need to set the version from a manifest file? Can't it always use the version in the branch?

Reply:
Do you mean to hardcode `7.17`?

Reply:
No I don't understand what this manifest file block is doing. Is this something that applies to all DRA built projects?

Mostly I want to double check that this isn't something from the elastic-agent repository that was done support early agent releases that isn't actually needed here. The early agent releases work flow can't work for 7.17 as it is missing the changes the agent itself needed to support it.

Reply:
It was introduced before the faster agent release project was started. If I understand correctly, it get the version from a provided manifest(specified in build params) and just sets it to `BEAT_VERSION`

Reply:
Got it, it came from here and is used to know which dependency versions to download. https://github.com/elastic/elastic-agent/pull/2727


I still don't really understand why it sets BEAT_VERSION since I doubt that ever doesn't match the version in the branch.